lms/djangoapps/instructor_task/tasks_helper/grades.py

```diff
@@ -519,131 +644,135 @@ def _rows_for_users(self, context, users):
             return success_rows, error_rows
 
 
-class ProblemGradeReport(object):
+class ProblemGradeReport(GradeReportBase):
+    """
+    Class to encapsulate functionality related to generating Problem Grade Reports.
+    """
     @classmethod
     def generate(cls, _xmodule_instance_args, _entry_id, course_id, _task_input, action_name):
+        """
+        Public method to generate a grade report.
+        """
+        with modulestore().bulk_operations(course_id):
+            context = _ProblemGradeReportContext(_xmodule_instance_args, _entry_id, course_id, _task_input, action_name)
+            # pylint: disable=protected-access
+            return ProblemGradeReport()._generate(context)
+
+    def _generate(self, context):
         """
         Generate a CSV containing all students' problem grades within a given
         `course_id`.
         """
+        context.update_status('Starting grades')
+        course = get_course_by_id(context.course_id)
+        header_row = OrderedDict([('id', 'Student ID'), ('email', 'Email'), ('username', 'Username')])
+        graded_scorable_blocks = self._graded_scorable_blocks_to_header(course)
+        success_headers = self._success_headers(header_row, graded_scorable_blocks)
+        error_headers = self._error_headers(header_row)
 
-        def log_task_info(message):
-            """
-            Updates the status on the celery task to the given message.
-            Also logs the update.
-            """
-            fmt = u'Task: {task_id}, InstructorTask ID: {entry_id}, Course: {course_id}, Input: {task_input}'
-            task_info_string = fmt.format(
-                task_id=task_id, entry_id=_entry_id, course_id=course_id, task_input=_task_input
-            )
-            TASK_LOG.info(u'%s, Task type: %s, %s, %s', task_info_string, action_name, message, task_progress.state)
+        generated_rows = self._batched_rows(context, header_row, graded_scorable_blocks)
+        success_rows, error_rows = self._compile(context, generated_rows)
+        self._upload(context, success_headers, success_rows, error_headers, error_rows, 'problem_grade_report')
```

Comment:
let's define `problem_grade_report` in the context class and use it here as `context.file_name`